src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -2134,6 +2149,33 @@ public override bool HandleCall(MethodIL callingMethodBody, MethodDesc calledMet
                         break;
 
                     default:
+                        if (calledMethod.IsPInvoke)
+                        {
+                            // Is the PInvoke dangerous?
+                            ParameterMetadata[] paramMetadata = calledMethod.GetParameterMetadata();
+
+                            ParameterMetadata returnParamMetadata = Array.Find(paramMetadata, m => m.Index == 0);
+
+                            bool comDangerousMethod = IsComInterop(returnParamMetadata.MarshalAsDescriptor, calledMethod.Signature.ReturnType);
```

Comment:
NativeAOT has built-in COM interop unconditionally unsupported. How does this work with built-in COM interop supported switch? (`System.Runtime.InteropServices.BuiltInComInterop.IsSupported` in CoreCLR.)

Reply:
This always warns in IL Linker, irrespective of the switch value.